src/zjs_promise.c

```diff
@@ -46,13 +46,6 @@ static void post_promise(void* h, jerry_value_t* ret_val)
         jerry_release_value(handle->then);
         jerry_release_value(handle->catch);
         jerry_release_value(handle->this);
-    }
-}
-
-static void promise_free(const uintptr_t native)
-{
-    struct promise* handle = (struct promise*)native;
-    if (handle) {
         zjs_free(handle);
```

Comment:
Seems like maybe you should also remove the object from the "promise" property of the parent object? Like set that to undefined? Also maybe that "promise" property should be "hidden", however you did that magic? Please look into those questions but I won't hold up the patch for it.